crossbar/router/broker.py

```diff
@@ -171,13 +171,17 @@ def _filter_publish_receivers(self, receivers, publish):
 
         # if "eligible_authid" we only accept receivers that have the correct authid
         if publish.eligible_authid:
+            eligible = set()
             for aid in publish.eligible_authid:
-                receivers = receivers & self._router._authid_to_sessions.get(aid, set())
```

Comment:
(as in the "red" code that's currently already merged on master is the wrong stuff)